packages/frontend/editor-ui/src/stores/nodeTypes.store.ts

```diff
@@ -262,14 +275,33 @@ export const useNodeTypesStore = defineStore(STORES.NODE_TYPES, () => {
 		return nodesInformation;
 	};
 
-	const getFullNodesProperties = async (nodesToBeFetched: INodeTypeNameVersion[]) => {
+	const getFullNodesProperties = async (
+		nodesToBeFetched: INodeTypeNameVersion[],
+		replaceNodeTypes = true,
+	) => {
 		const credentialsStore = useCredentialsStore();
 		await credentialsStore.fetchCredentialTypes(true);
-		await getNodesInformation(nodesToBeFetched);
+		if (replaceNodeTypes) {
+			await getNodesInformation(nodesToBeFetched);
+		}
 	};
 
 	const getNodeTypes = async () => {
-		const nodeTypes = await nodeTypesApi.getNodeTypes(rootStore.baseUrl);
+		let nodeTypes = await nodeTypesApi.getNodeTypes(rootStore.baseUrl);
+		let attempts = 5;
+
+		while (typeof nodeTypes !== 'object' && attempts > 0) {
```

Comment:
Why do we need this retry mechanism?

Reply:
And if it is absolutely needed, it should have some delay, you could maybe use `useReconnectTimer`

Reply:
in some cases getNodeTypes returns a string, probably due to lazy loading, this needs farther investigation and may not be needed in the future, also added delay as bot suggested